core/src/main/java/org/apache/spark/api/shuffle/ShufflePartitionWriter.java

```diff
@@ -34,9 +33,7 @@
 public interface ShufflePartitionWriter {
   OutputStream openStream() throws IOException;
 
-  long getLength();
+  long closeAndGetLength();
 
-  default WritableByteChannel openChannel() throws IOException {
-    return Channels.newChannel(openStream());
-  }
+  FileChannel openChannel() throws IOException;
```

Comment:
This should still return `WritableByteChannel`, but the implementation returns `FileChannel` in `DefaultShufflePartitionWriter`.

Reply:
The code in `Utils#transferTo` will eventually check `instanceOf` this `WritableByteChannel` and optimize on `FileChannel` accordingly,